topics/go/algorithms/numbers/reverse/reverse.go

```diff
@@ -0,0 +1,24 @@
+package numbers
+
+// Reverse takes in a integer and returns a the integer in reverse order.
+func Reverse(num int) int {
+
+	// Set result to zero.
+	var result int
+
+	// Loop until num is not zero.
+	for num != 0 {
+
+		// Get the last digit from num.
+		// Example: 125 % 10 = 5.
+		last := num % 10
+
+		// Move result one place to the left and add last.
+		result = result*10 + last
+
+		// Remove the right most digit from num.
+		// Example: 125 / 10 = 12.
+		num = num / 10
+	}
+	return result
```

Comment:
I think I want to find a way to show the example so people can see how the algorithm works. Maybe we show each interation 
https://play.golang.org/p/Gy33iB6wGs0

last[5] = num[125] % 10
last[2] = num[12] % 10
last[1] = num[1] % 10

result[0] = result[0] * 10
result[50] = result[5] * 10
result[520] = result[52] * 10

result[5] = result[0] += last[5]
result[52] = result[50] += last[2]
result[521] = result[520] += last[1]

num[12] = num[125] / 10
num[1] = num[12] / 10
num[0] = num[1] / 10

Reply:
Should we create a main.go to show this? Or include it in reverse.go?

Reply:
Somehow in this file or we add a doc.go